docs/architecture.md

```diff
@@ -0,0 +1,473 @@
+        end
+        subgraph SS[Synchronizer Service]
+            SSD{{Domain}}
+        end
+        subgraph P2PS[P2P Service]
+            P2PD{{Domain}}
+            POAP2P[PoA Adapter]
+            SYNCP2P[Sync Adapter]
+            TXP2P[Transaction Pool Adapter]
+        end
+        subgraph TXS[Transaction Pool Service]
+            TXD{{Domain}}
+            BPTXS[Block Producer Adapter]
+            BITXS[Block Importer Adapter]
+        end
+        subgraph RS[Relayer Service]
+            RSD{{Domain}}
+            POAR[POA Adapter]
+            BPR[Block Producer Adapter]
+            BIR[Block Importer Adapter]
+        end
+        subgraph ES[Executor Service]
+            ESD{{Domain}}
+            BPEXE[Block Producer Adapter]
+            BIEXE[Block Importer Adapter]
+        end
+        subgraph APIS[API Service]
+            APID{{Domain}}
+        end
+        BID & BPD --> ES & RS & TXS
+        SSD --> POAS & P2PS & BIS
+        TXD --> P2PS
+        POAD --> P2PS & RS & BPS & BIS
+        APID --> BPS
+        
+        %% Red color - Block Importer
+        linkStyle 0,1,2 fill:none,stroke-width:4px,stroke:red
+        %% Pink color - Block Producer
+        linkStyle 3,4,5 fill:none,stroke-width:4px,stroke:pink
+        %% Green color - Synchronizer
+        linkStyle 6,7,8 fill:none,stroke-width:4px,stroke:green
+        %% Blue color - Transaction pool
+        linkStyle 9 fill:none,stroke-width:4px,stroke:blue
+        %% Orange color - PoA
+        linkStyle 10,11,12,13 fill:none,stroke-width:4px,stroke:orange
+        %% Purple color - API
+        linkStyle 14 fill:none,stroke-width:4px,stroke:purple
+    end
+```
+
+### (PoA) Consensus Module
+
+#### Ports: fuel_poa_consensus::ports
+```rust
+trait BlockProducer {
+    // used by the PoA node to produce a new block and obtain fully computed block headers
+    async fn produce_block(height: BlockHeight) -> Result<FuelBlock>;
+}
+
+trait BlockImporter {
+    // After the PoA node has signed over the block, send it to the block importer for inclusion
+    // into the finalized state of the node.
+    async fn commit(sealed_block: SealedFuelBlock) -> Result<()>;
+}
+
+trait Relayer {
+    // wait until the relayer eth node is synced
+    async fn await_synced() -> Result<()>;
+    // wait until the relayer reaches a desired da_height
+    async fn await_da_height(da_height: DaHeight) -> Result<()>;
+    // get the consensus key associated with a da_height
+    async fn get_poa_key(da_height: DaHeight) -> Result<ConsensusKey>;
+}
+
+trait Database {
+    // get the current fuel chain height
+    fn get_current_height() -> Result<BlockHeight>;
+    // insert consensus information associated with a sealed block that was locally produced
+    fn insert_consensus_data(sealed_block: &SealedFuelBlock) -> Result<()>;
+}
+
+trait PeerToPeer {
+    // broadcast a newly produced block to the network
+    async fn broadcast_new_block(block: SealedFuelBlock) -> Result<()>;
```

Comment:
Might be worth just adding a comment as to what this function actually awaits. Broadcasts are traditionally best effort but I think the options are:
1. Await a successful connection to N peers.
2. Await a successful response "received" response from N peers.
3. Await on a permit to send a broadcast (to avoid overloading the background broadcast task).
4. Don't await anything (probably not recommended as this acts like an unbounded channel then).
It might depend on the under lying lib on what we can await. IMHO it's best to go option `3.` as `1.` or `2.` can create make this function very slow to return (imagine talking to a peer on a crappy network connection) and it might not be in a part of the system where you want that.

Reply:
I think we can actually remove this broadcast, we will use the broadcast from the block importer for PoA instead.

Reply:
I think the same question applies to that broadcast as well.